src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -66,6 +47,12 @@ public Assembly LoadFromPath(string fullPath)
 
             lock (_guard)
             {
+                if (!_resolvingHandlerHookedUp)
+                {
+                    AssemblyLoadContext.Default.Resolving += TryResolveAssembly;
```

Comment:
@cdmihai If we could, it would be useful to be able to return an msbuild process to some sort of "clean" state, where task assemblies were unloaded and the event handler removed. As we cannot currently unload the assemblies, removing the event handler isn't that interesting.